### PR TITLE
ft user requests table 167735047

### PR DESCRIPTION
**Pull request by SamuelKoroh**
**What does this PR do?**

 Integrate the requests table to the backend by retrieving all requests made by a user

**Description of Task to be completed?**
- setup my-requests route
- setup the method in the requestController to handle the request
- setup the method in requestService to retrieve the user request
- document the end point in swagger
- add the api end point to the read me
  
**How should this be manually tested?**
- from swagger under Request category select GET/requests​/my-requests
- provide valid user token then execute

**What are the relevant stories?**

https://www.pivotaltracker.com/story/show/167735047

**Screenshots (if appropriate)**
image

### CODE REVIEW

**Review by prieste — APPROVED** [review]

File: src/tests/testData/sampleData.js [hunk 1]
```diff
@@ -2,16 +2,33 @@ export const multiRequest = {
...
   tripType: 'multi-city',
-  travelDate: '2038-01-19 03:14:07 ',
-  returnDate: '2038-01-19 03:14:07',
+  travelDate: Date.now(),
+  returnDate: Date.now(),
   reason: 'reason',
   accommodation: 'accommodation'
 };
 
 export const missingRequiredField = {
```
Comment (hameedhub) [inline src/tests/testData/sampleData.js:11]: travelDate is a required field, i think you should check is entered

File: src/tests/request.test.js [hunk 2]
```diff
@@ -241,4 +239,28 @@ describe('/api/v1/requests', () => {
...
     });
   });
+  describe('GET /', () => {
+    it('should retrieve all requests made by the users', async () => {
+      const res = await chai.request(app)
+        .get(`${URL_PREFIX}/my-requests`)
+        .set('token', loginUser.body.data.token);
+
+      res.should.have.status(200);
+      res.body.data[0].should.have.property('destination');
```
Comment (hameedhub) [inline src/tests/request.test.js:249]: you could also assert the property value e.g `res.body.data[0].should.have.property('destination').eql('Lagos')` to be sure we are not getting a `null` value

**Review by hameedhub — COMMENTED** [review]
LGTM

File: src/tests/request.test.js (hunk 2 above)
Comment (SamuelKoroh) [reply src/tests/request.test.js:249]: Ok

File: src/tests/testData/sampleData.js (hunk 1 above)
Comment (SamuelKoroh) [reply src/tests/testData/sampleData.js:11]: Ok

**Review by amoskeyz — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -42,4 +42,5 @@ docker run -p 3000:3000 iceman-backend
 <tr><td>GET</td><td>/api/v1/auth/google</td><td>Authenticate user using Google</td></tr>
 <tr><td>POST</td><td>/api/v1/requests/multi-city</td><td>Allow user to make multi city request</td></tr>
 <tr><td>PATCH</td><td>/api/v1/request/:id</td><td>Updates trip requests</td></tr>
+<tr><td>GET</td><td>/api/v1/request/my-requests</td><td>Retrieve user requests</td></tr>
```
Comment (amoskeyz) [inline README.md]: it should be `requests/my-request` not `request/my-request`

File: src/controllers/requestController.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import RequestService from '../services/requestService';
 import Response from '../utils/response';
 
 const { success, badRequest } = Response;
-const { updateRequest, multiCityRequest } = RequestService;
+const { updateRequest, multiCityRequest, myRequests } = RequestService;
```
Comment (Annmary12) [inline src/controllers/requestController.js]: instead of `myRequests`, change it to `getRequests`

File: src/controllers/requestController.js [hunk 5]
```diff
@@ -53,4 +53,19 @@ export default class RequestController {
       badRequest(res, err);
     }
   }
+
+  /**
+   * @param {req} req - request obiect
+   * @param {res} res - res object object
+   * @returns {object} - return object
+   */
+  static async myRequests(req, res) {
```
Comment (Annmary12) [inline src/controllers/requestController.js]: Change `myRequests` to `getRequests`

File: src/routes/request.js [hunk 6]
```diff
@@ -6,10 +6,16 @@ import middlewares from '../middlewares';
 
 const router = Router();
 const { auth } = middlewares;
-const { update, oneWay, multiCityRequest } = RequestController;
+const {
+  update,
+  oneWay,
+  multiCityRequest,
+  myRequests
```
Comment (Annmary12) [inline src/routes/request.js]: The same here, change `myRequests` to `getRequests`

File: src/routes/request.js [hunk 7]
```diff
@@ -6,10 +6,16 @@ import middlewares from '../middlewares';
...
+} = RequestController;
 
 router.post('/multi-city', [auth, validator(requestSchema)], multiCityRequest);
 router.post('/one-way', [auth, validator(requestSchema)], oneWay);
 router.patch('/:requestId', [auth, validator(requestSchema)], update);
+router.get('/my-requests', auth, myRequests);
```
Comment (Annmary12) [inline src/routes/request.js]: Remove `my-requeste`  from the route and leave it as `/`

File: src/services/requestService.js [hunk 8]
```diff
@@ -54,16 +54,30 @@ export default class RequestService {
...
+    const { dataValues } = await Request.create({ ...body, destination, userId: id });
 
     return dataValues;
   }
+
+  /**
+   * @param {object} body - arrays of request object
+   * @returns {object} obej - return object
+   */
+  static async myRequests({ user: { id } }) {
```
Comment (Annmary12) [inline src/services/requestService.js]: Change `myRequests` to `getRequests`

**Pull request merged**